frontend/src/pages/DashboardPage/DashboardPage.tsx

```diff
@@ -1,39 +1,75 @@
 import { Button } from "@/components/ui/button";
 import { useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
 import { DISCOVER_PATH } from "@/lib/paths";
 import StreamerCard from "@/components/StreamerCard/StreamerCard";
+import { fetchStreamerCardInfo } from '@/lib/utils';
+import { useAuthContext } from "@/Context/AuthProvider";
 
 export default function DashboardPage() {
   const navigate = useNavigate();
+  const { session } = useAuthContext();
+  const [userProfile, setUserProfile] = useState(null);
+  const [loading, setLoading] = useState(true);
 
-  //keep in mind in future below discover streamers button wont be positioned absolutely but rather be in a flex box container that also contains prompt of day and recent activity
-  return (
-    
-    <div className = "flex flex-row gap-1 p-14 min-h-screen">
+  useEffect(() => {
+
+    async function populateStreamerCard(){
+      if (!session || !session.access_token) {
+        console.error("No valid session");
+        //if there isnt a session we set loading to false because we currently arent checking profile
+        setLoading(false); 
+        return;
```

Comment:
You can move this inside the try-catch and directly throw an error here, you're already handling the errors there 👀